src/wurf/dependency.py

```diff
@@ -79,34 +91,52 @@ def __init__(self, **kwargs):
         #    in external formats such as files
         # 2. Use the Python hash() function to generate a integer hash for 
         #    run-time usage (i.e. within the same interpreter session).
-        self.hash = None
+        #super(Dependency, self).__setattr__('hash', None)
+        self.read_only['hash'] = None
+        
+        # Resolver attributes (modifiable)
+        object.__setattr__(self, 'read_write', dict())
+        
         
     def __getattr__(self, option):
         """ Return the value corresponding to the option.
 
         :param option: The name of the option to return as a string.
         :return: The option value.
         """
+        if option in self.read_only:
+            return self.read_only[option]
+
+        elif option in self.read_write:
+            return self.read_write[option]
+        
+        else:
+            return None
+
+
+    def __setattr__(self, name, value): 
+        """ Sets a resolver attribute."""
 
-        if option in self.info:
-            return self.info[option]
+        if name in self.read_only:
+            raise AttributeError("Attribute {} read-only.".format(name))
         else:
-            raise AttributeError("No such attribute: " + option)
+            self.read_write[name] = value   
 
     def __contains__(self, option):
         """ Checks if the option is available.
         
         :return: True if the option is available otherwise False
         """
-        return option in self.info
+        return (option in self.read_only) or (option in self.read_write)
 
     def __str__(self):
-        return str(self.info)
+        return "Dependency\ninfo: {}\nresolver info: {}".format(
+            self.read_only, self.read_write)
 
     def __hash__(self):
         """ :return: Integer hash value for the dependency. """
         
-        if not self.hash:
-            self.hash = hash(self.info['sha1'])
+        if not self.read_only['hash']:
+            self.read_only['hash'] = hash(self.read_only['sha1'])
```

Comment:
I am just curious: Do we use this hash value for anything besides comparing 2 dependencies in dependency_manager?:
`if seen_dependency.sha1 != dependency.sha1:`
They are not stored in any dict with their hash value, right?

If the comparison is the only use case, then I think that the hash might be too strict down the line. Just consider all the recursion that happens in our projects. If there is a slight mismatch, the comparison fails. 
For example, let's say that I add a new source repo to a given dependency, because we moved it to a new location. Then I have to make the same modification in all the wscripts where it is used, otherwise I get a mismatch.

This is beyond the scope of this PR, but I think we need to consider a smarter way of determining compatibility.